### PR TITLE
Add textarea variant to checkbox component

### LINKED ISSUE #3402: Checkbox with revealed text area
As Software Developer
I want to be able to add checkbox with the revealed text area
So that users responding "Other" can provide the legally required justification.

**Description**
This request came from the Community discussion board - [Checkbox](https://github.com/ONSdigital/design-system/discussions/13#discussioncomment-10902568)

Due to the tightly regulated format of the form, whenever respondent selects "Other" option, they need to provide up to 700 words (3500 characters) justification. However this field should be editable - users can adjust the number of characters as per their individual needs.
The characters count needs to be added too.  It will be counting down 'characters remaining', like all other existing design system components.
Text area will need to be able to accommodate the text size. User expects the text area to expand horizontally to the limit of it's containing parent/grid element + obey existing Nunjucks size attributes for textarea. If the content then gets too big for textarea to display then should only display a vertical scrollbar. 

**Acceptance Criteria**
  Scenario: Populate textarea when "other" checkbox is selected
    Given the user is on the form page
    When the user selects the checkbox called "other"
    Then the textarea is revealed 

  Scenario: Limit the number of characters user can add
    Given the user is on the form page
    And the textarea is populated
    When the user tries to add characters
    Then the user can add a maximum number of characters as stated under the textarea

  Scenario: Trigger remaining character count on any input
    Given the user is on the form page
    And the textarea is populated
    When the user inputs characters
    Then the remaining character count is triggered

  Scenario: Allow user to input and delete characters
    Given the user is on the form page
    And the textarea is populated
    When the user inputs characters
    And the user deletes characters
    Then the textarea reflects the input and deletion of characters

  Scenario: Adjust textarea size based on the number of characters
    Given the user is on the form page
    And the textarea is populated
    When the user inputs characters
    Then the textarea size expands horizontally to the limit of it's containing parent/grid element 

  Scenario: Scroll bar within textarea size based on the number of characters
    Given the user is on the form page
    And the textarea is populated
    When the user inputs large number of characters
Then the textarea will display vertical scrollbar

**Visualisation/Input from UCD**
This is the visualisation of the requirement provided by the requestor
Image

**Linked User stories/epics**
Currently we have similar functionality in the 
1/ [Checkbox error message](https://service-manual.ons.gov.uk/design-system/components/checkboxes/example-checkboxes-error) where user can select other and it reveals textarea.
2/ [Radios with reveled textarea](https://service-manual.ons.gov.uk/design-system/components/radios/example-radios-with-revealed-text-input) 

Technical details (optional)
**Delivery expected by end of November**

### PR DESCRIPTION
**Pull request by precious-onyenaucheya-ons**
### What is the context of this PR?

Fixes #3402 

This PR introduces a new variant of the Checkboxes component. The new variant includes an integrated text area that allows users to input text, with a character limit checker to ensure the input doesn't exceed the specified limit.

### How to review this PR

- Check that text area variant has been added with a character limit checker
- Check that all unit and visual tests pass

### Checklist

This needs to be completed by the person raising the PR.

-   [x] I have selected the correct Assignee
-   [x] I have linked the correct Issue

### CODE REVIEW

File: src/components/checkboxes/example-checkboxes-with-text-area-and-character-limit.njk [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                },
+                {
+                    "id": "other-checkbox-with-revealed-text-input",
+                    "label": {
+                        "text": "Other"
+                    },
+                    "value": "other",
+                    "other": {
+                        "otherType": "textarea",
+                        "id": "other-textbox-example-checkbox-with-revealed-text-area",
```
Comment (rmccar) [inline src/components/checkboxes/example-checkboxes-with-text-area-and-character-limit.njk:52]: This id should either match the others and have `text-input` on the end or they should all be `text-area`

File: src/components/checkboxes/example-checkboxes-with-text-area-and-character-limit.njk [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                        "name": "other-answer",
+                        "label": {
+                            "text": "Provide more details"
+                        },
+                         "charCheckLimit": {
+                            "limit": 400,
```
Comment (rmccar) [inline src/components/checkboxes/example-checkboxes-with-text-area-and-character-limit.njk:58]: The radio textarea examples set this to 200, should probably make that consistent

File: src/components/checkboxes/_macro.spec.js [hunk 3]
```diff
@@ -135,6 +160,15 @@ describe('macro: checkboxes', () => {
         });
     });
 
+    describe('checkbox item with text area', () => {
```
Comment (rmccar) [inline src/components/checkboxes/_macro.spec.js:163]: We should text the maxlength attribute here to be set to what the char limit should be

**Review by rmccar — CHANGES_REQUESTED** [review]
Few minor things, other than this looks good

**Review by rmccar — CHANGES_REQUESTED** [review]

File: src/components/checkboxes/example-checkboxes-with-revealed-text-area-and-character-limit-expanded.njk [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                        "id": "other-textbox-example-checkbox-with-revealed-text-area",
+                        "name": "other-answer",
+                        "label": {
+                            "text": "Provide more details"
+                        },
+                         "charCheckLimit": {
+                            "limit": 200,
+                            "charCountSingular": "You have {x} character remaining",
+                            "charCountPlural": "You have {x} characters remaining"
+                        }
```
[the remark this answers is not in the document]
Comment (rmccar) [reply src/components/checkboxes/example-checkboxes-with-revealed-text-area-and-character-limit-expanded.njk]: This line still needs to be formatted

**Comment by rmccar** [comment]
Just need to remove the old vr screenshots from the PR now

**Review by rmccar — CHANGES_REQUESTED** [review]

**Review by adi-unni — APPROVED** [review]

**Review by rmccar — APPROVED** [review]

**Pull request merged**